qiskit_experiments/test/mock_iq_backend.py

```diff
@@ -91,5 +94,57 @@ def _draw_iq_shot(self, prob) -> List[List[float]]:
         else:
             return [[self._iq_cluster_centers[2] + rand_i, self._iq_cluster_centers[3] + rand_q]]
 
-    def run(self, run_input, **options) -> TestJob:
-        """Subclasses will need to override this."""
+    @abstractmethod
+    def _compute_probability(self, circuit: QuantumCircuit) -> float:
+        """Compute the probability used in the binomial distribution creating the IQ shot.
+
+        An abstract method that subclasses will implement to create a probability of
+        being in the excited state based on the received quantum circuit.
+
+        Args:
+            circuit: The circuit from which to compute the probability.
+
+        Returns:
+             The probability that the binaomial distribution will use to generate an IQ shot.
+        """
+
+    # pylint: disable=arguments-differ
+    def run(self, circuits, shots=1024, meas_level=MeasLevel.KERNELED, meas_return="single"):
+        """Run the spectroscopy backend."""
+
+        result = {
+            "backend_name": f"{self.__class__.__name__}",
+            "backend_version": "0",
+            "qobj_id": 0,
+            "job_id": 0,
+            "success": True,
+            "results": [],
+        }
+
+        for circ in circuits:
+            run_result = {
+                "shots": shots,
+                "success": True,
+                "header": {"metadata": circ.metadata},
+            }
+
+            prob = self._compute_probability(circ)
+
+            if meas_level == MeasLevel.CLASSIFIED:
+                counts = {"1": 0, "0": 0}
+
+                for _ in range(shots):
+                    counts[str(self._rng.binomial(1, prob))] += 1
```

Comment:
Perhaps this for loop can be replaced with a single call to `binomial`, something like `binomial(shots, prob)`. Maybe also in other experiments.

Reply:
That makes sense.  See here: 4a6040b